Solutions/Azure Activity/Analytic Rules/NewResourceGroupsDeployedTo.yaml

```diff
@@ -17,34 +17,29 @@ tactics:
 relevantTechniques:
   - T1496
 query: |
-  let szOperationNames = dynamic(["Microsoft.Compute/virtualMachines/write", "Microsoft.Resources/deployments/write"]);
+  // Add or remove operation names below as per your requirements. For operations lists, please refer to https://learn.microsoft.com/en-us/Azure/role-based-access-control/resource-provider-operations#all
+  let szOperationNames = dynamic(["Microsoft.Compute/virtualMachines/write", "Microsoft.Resources/deployments/write", "Microsoft.Resources/subscriptions/resourceGroups/write"]);
   let starttime = 14d;
   let endtime = 1d;
   let RareCaller = AzureActivity
   | where TimeGenerated between (ago(starttime) .. ago(endtime))
   | where OperationNameValue in~ (szOperationNames)
-  | project ResourceGroup, Caller, OperationNameValue, CallerIpAddress
+  | summarize ActivityStatusValue = make_set(ActivityStatusValue), ResourceId = make_set(_ResourceId), OperationNameValue=make_set(OperationNameValue), CallerIpAddress = make_set(CallerIpAddress), ResourceGroup=make_set(ResourceGroup) by Caller
+  // Returns all the records from the right side that don't have matches from the left.
   | join kind=rightantisemi (
   AzureActivity
   | where TimeGenerated > ago(endtime)
   | where OperationNameValue in~ (szOperationNames)
-  | summarize StartTimeUtc = min(TimeGenerated), EndTimeUtc = max(TimeGenerated), ActivityStatusValue = makeset(ActivityStatusValue), OperationIds = makeset(OperationId), CallerIpAddress = makeset(CallerIpAddress) 
-  by ResourceId, Caller, OperationNameValue, Resource, ResourceGroup
-  ) on Caller, ResourceGroup 
-  | mvexpand CallerIpAddress
-  | where isnotempty(CallerIpAddress);
-  let Counts = RareCaller | summarize ActivityCountByCaller = count() by Caller;
-  RareCaller | join kind= inner (Counts) on Caller | project-away Caller1
-  | extend timestamp = StartTimeUtc, AccountCustomEntity = Caller, IPCustomEntity = tostring(CallerIpAddress)
-  | sort by ActivityCountByCaller desc nulls last 
+  | summarize StartTimeUtc = min(TimeGenerated), EndTimeUtc = max(TimeGenerated), ActivityStatusValue = make_set(ActivityStatusValue), ResourceId = make_set(_ResourceId), OperationNameValue=make_set(OperationNameValue), CallerIpAddress = make_set(CallerIpAddress), ResourceGroup=make_set(ResourceGroup) by Caller
+  ) on Caller ;
+  RareCaller
+  | extend Name = tostring(split(Caller,'@',0)[0]), UPNSuffix = tostring(split(Caller,'@',1)[0])
```

Comment:
Might not be a UPN - could be an SP ID - we need to account for this.

Reply:
Same case as previous one.

Reply:
Same as above

Reply:
done